docs/contributing/999-rfcs/2023-10-01-workloads.md

```diff
@@ -0,0 +1,252 @@
+---
+title: "#4350 Containerized Workloads - Requirements"
+description: Requirements for containerized workloads in the Wing Cloud Library
+---
+
+# Containerized Workloads - Requirements Specification
+
+- **Author(s)**: @eladb
+- **Submission Date**: 2023-10-01
+- **Stage**: Proposal
+- **Stage Date**: 2023-10-01
+
+This specification lists the *requirements* we have identified so far that are related to using
+containers to implement long running workloads (aka "services" or "microservices"). 
+
+## Requirements
+
+Requirements are marked with a *Reqtag* so they will be easy to trace in subsequent design
+documents. Suggested priorities are marked inline as "(P1)", "(P2)" or "(P3)".
+
+### Containerized workloads (P1)
+
+> Reqtag: `c:workload`
+
+Wing must support defining **containerized workloads** implemented through user-defined container
+images. Images may either be pulled from a container registry or built from a `Dockerfile` within
+the project directory.
```

Comment:
What's the relationship between "Workload", "Container", and "Function". Can I describe a container workload to run in a Function?

Reply:
The use case of running containers within serverless platforms is currently out of scope for this RFC, as well as using any other runtime within serverless functions. We'll scope that out separately.

I'll add a note.

As for the terminology, I'll add some definitions at the top.

Reply:
> The use case of running containers within serverless platforms is currently out of scope for this RFC

Do you just mean within FaaS platforms, or outside of any managed container platforms? I've heard some folks call ECS serverless, but honestly even I get easily confused sometimes 😅